pkgs/tools/admin/awscli2/default.nix

```
@@ -21,6 +21,7 @@ let
       });
       urllib3 = prev.urllib3.overridePythonAttrs (prev: {
         format = "setuptools";
+        pyproject = null;
```

This shouldn't be null. Check https://github.com/NixOS/nixpkgs/pull/254136
```suggestion
        pyproject = true;
```

---

I fixed it locally using the below but like the above comment from @WolfangAukang better.
```
        # format = "setuptools";
        pyproject = true;
        nativeBuildInputs = [
          final.setuptools
        ];
```

---

In fact I think yours is the correct approach as it is inside an overlay @vtumuluri

---

"adds the pyproject option that conflicts with format, where pyproject = true is equivalent to format = "pyproject""
We weren't using pyproject previously, that changed here https://github.com/NixOS/nixpkgs/commit/322442238cd7bed899e774f25b62c305fc9dc519. Are you certain we wanna change from setuptools to pyproject?

---

> Are you certain we wanna change from setuptools to pyproject?

As stated on the [docs](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters):

> pyproject: Whether the pyproject format should be used. When set to true, pypaBuildHook will be used, and you can add the required build dependencies from build-system.requires to nativeBuildInputs. Note that the pyproject format falls back to using setuptools, so you can use pyproject = true even if the package only has a setup.py.

And also there is [this comment](https://github.com/NixOS/nixpkgs/pull/254136#discussion_r1320594064):

> by that I mean pypaBuildHook will try to use setuptools even if pyproject.toml doesn't exist, but it still needs to be added to nativeBuildInputs

I know it is confusing, but it seems that is the new approach now.

---

Thanks